### PR TITLE
test(mobile): add comprehensive onboarding flow E2E test with error cases

### PR DESCRIPTION
**Pull request by compojoom**
## What it solves
The test is similar to other onboarding flow, but we also test some edge cases here 

Resolves https://linear.app/safe-global/issue/COR-726/test-018-account-import-manual-entry-validation

## Screenshots

https://github.com/user-attachments/assets/b576c5c7-5a00-4e35-880c-d8d628677a68

## Checklist

- [ ] I've tested the branch on mobile 📱
- [ ] I've documented how it affects the analytics (if at all) 📊
- [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

---

## CLA signature

With the submission of this Pull Request, I confirm that I have read and agree to the terms of the [Contributor License Agreement](https://safe.global/cla).

### CODE REVIEW

File: apps/mobile/e2e/tests/onboarding/enhanced-onboarding-flow.yml [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+# Clear input and enter valid EOA address (not a Safe)
+- eraseText
+- inputText: 'sep:0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266'
+
+# Wait for validation to complete
+- extendedWaitUntil:
+    visible:
+      text: '.*No Safe deployment found for this address.*'
+    timeout: 5000
```
[the remark this answers is not in the document]
Comment (compojoom) [reply apps/mobile/e2e/tests/onboarding/enhanced-onboarding-flow.yml:85]: We are waiting for a network request here and not for elements that are by default visible/not visible depending on data we already have. That's why I've opted to hardcode the same value for both iOS and Android
